RELEASE_NOTE_DRAFT.md

```diff
@@ -1,25 +1,102 @@
+
+# Optimizations
+----------------------
+* Optimized ImmutableListFactoryImpl.withAll(Iterable).
+* Optimized containsAll on primitive iterables.
+* Optimized primitive hash set newSet and withAll.
+* Optimized sumByLong and sumByInt Primitive methods for Bags.
+* Optimized toImmutable on primitive sets.
+* Optimized union and difference on SetIterables.
+* Added enhanced for-loop syntax wherever as possible.
 
 # Tech Debt Reduction
 ---------------------
-* 
-
-# OSGi Integration Improvements
--------------------------------
-* 
+* Deprecated  ImmutableArrayStack.
+* Fixed bug with noneSatisfy for ImmutableBooleanEmptySet.
+* Fixed compiler errors for JDK-15-EA.
+* Fixed return types of aggregateBy().
+* Fixed types on Multimap.*MultiValues().
+* Fixed primitive hash map values collections removed  for special key zero.
+* Fixed CollectIterable detect methods.
+* Added missing overrides for toImmutable methods on synchronized primitive object maps.
+* Added missing overrides of aggregateBy() and aggregateInPlaceBy().
+* Added documentation on serializing Eclipse Collections with jackson.
+* Added logic to throw UnsupportedOperationException on calling withMap method in FixedSizeMap. 
+* Made org.eclipse.collections.impl.list.Interval#goForward private.
+* Replaced HashingStrategies.longHashCode with Java 8 Long.hashCode.
+* Deleted version.properties as the file is no longer needed.
+* Removed extra calls to map.get in sumByDoubleFunction and sumByFloatFunction.
+* Removed implementations of OrderedIterable.toStack() that can use the default implementation instead.
+* Removed implementations of aggregateBy and aggregateInPlaceBy that can use default implementations instead.
+* Removed references to deleted classes in findbugs-exclude file.
+
+# Removed Functionality
+--------------------------
+* Removed Verify.assertThrows() which takes a Runnable in favor of Assert.assertThrows(). This is a breaking change.
+* Removed assertNotEquals() from Verify as Assert already has the same API.
 
 # Build Changes
 -----------------
-* 
+* Upgraded EBR plugin to 1.3.0-SNAPSHOT.
+* Upgraded actions/cache to 2.1.6.
+* Upgraded actions/upload-artifact to 2.2.4.
+* Upgraded antlr ST4 to 2.1.5.
+* Upgraded checkstyle plugin to 3.1.2.
+* Upgraded checkstyle to 8.42.
+* Upgraded codehaus maven plugin to 2.8.1.
+* Upgraded jacoco to v2.2.3.
+* Upgraded jmh-core to 1.33.
+* Upgraded maven-resources-plugin to 3.2.0.
+* Upgraded to JUnit 4.13.1.
+* Upgraded to setup-java v2.
+* Skip p2 repository module during EA builds and Java 15+ builds.
+* Enabled Dependabot v2.
+* Added security badge in README.md.
+* Added GitHub action to generate code coverage report.
+* Added JDK Early Access GitHub Actions.
+* Added badges pointing to javadoc.io.
+* Added badges to website.
+
+# Breaking Changes
+-------------------------
+Warning: These changes are already mentioned above. The list below might not be exhaustive, make sure to test your application and usages to verify.
```

Comment:
We should also have a line at the corresponding change saying Breaking Change.

Reply:
Done.